### PR TITLE
Improve pluralization in automation descriptions

### PR DESCRIPTION
**Pull request by bdr99**
## Proposed change
Improve the pluralization when generating the descriptions for conditions and actions.
 - `Test if any of 1 conditions matches` -> `Test if any of 1 condition matches`
 - `Test if 1 conditions match` -> `Test if 1 condition matches`
 - `Test if none of 1 conditions match` -> `Test if 1 condition does not match`
 - `Choose between 1 actions` -> `Choose between 1 action`
 - `Run 1 actions in parallel` -> `Run 1 action in parallel`
 - `Delay for 1 seconds` -> `Delay for 1 second`
 - `Delay for 1 days 0:00:00` -> `Delay for 1 day 0:00:00`
 - `Repeat an action 1 times` -> `Repeat an action 1 time`

I also fixed an issue with the type definition of `CountRepeat`. The count can be a number or a string since it accepts a template.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/automation_i18n.ts [hunk 1]
```diff
@@ -374,35 +374,35 @@ export const describeCondition = (
   if (condition.condition === "or") {
     const conditions = ensureArray(condition.conditions);
 
-    let count = "condition";
-
     if (conditions && conditions.length > 0) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:377]: You reversed this one now?

File: src/data/script_i18n.ts [hunk 2]
```diff
@@ -231,32 +231,37 @@ export const describeAction = <T extends ActionType>(
...
-        ? `for every item: ${ensureArray(config.repeat.for_each)
-            .map((item) => JSON.stringify(item))
-            .join(", ")}`
-        : ""
-    }`;
+
+    let base = "Repeat an action";
+    if ("count" in config.repeat) {
+      const count = config.repeat.count;
+      base += ` ${count} time${count === "1" || count === 1 ? "" : "s"}`;
```
Comment (bramkragten) [inline src/data/script_i18n.ts:250]:
```suggestion
      base += ` ${count} time${Number(count) === 1 ? "" : "s"}`;
```

**Comment by bramkragten** [comment]
We should start moving this to translations, and thus start using plural there.

File: src/data/automation_i18n.ts (hunk 1 above)
Comment (bdr99) [reply src/data/automation_i18n.ts]: Nice catch, fixed. I intended to reverse that condition but must have missed it.

**Comment by bdr99** [comment]
@bramkragten Yes, I agree that these strings should ideally be translatable. I'm just trying to fix the most obvious issues in the meantime.

**Comment by kukulich** [comment]
As a translator I vote for translations.

**Comment by bdr99** [comment]
Again, I'm not against the idea of translations, but I suspect it is going to be a bit difficult to make these descriptions translatable. With this PR I'm just trying to improve the current situation while we work to achieve that goal.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**